src/services/common.ts

```diff
@@ -67,15 +67,17 @@ export class CommonService {
     const { asset, account: { finId } } = request;
     const { assetId } = assetFromAPI(asset);
     const balance = await this.finP2PContract.balance(assetId, finId);
-    const truncated = truncateDecimals(balance, this.defaultDecimals);
+    const held = await this.finP2PContract.held(assetId, finId);
```

Comment:
The fact we call balance and held independently can cause those 2 balances not aggregate to a correlated total. Do we need to get those 2 numbers with a single call to the ledger to make sure they are aligned?

Reply:
You're right. I think I need to refactor slightly the balance method altogether.